### PR TITLE
Navigation component setup

### PR DESCRIPTION
**Pull request by OmarMarei95**
Setting up the Navigation Component for managing the navigation flow between screens and define navigation graph.

### CODE REVIEW

File: app/src/main/res/layout/activity_main.xml [hunk 1]
```diff
@@ -6,13 +6,13 @@
...
-      app:layout_constraintBottom_toBottomOf="parent"
-      app:layout_constraintStart_toStartOf="parent"
-      app:layout_constraintEnd_toEndOf="parent"
-      app:layout_constraintTop_toTopOf="parent" />
+    <androidx.fragment.app.FragmentContainerView
+        android:id="@+id/fragmentContainer"
+        android:name="androidx.navigation.fragment.NavHostFragment"
+        android:layout_width="match_parent"
+        android:layout_height="match_parent"
+        android:visibility="visible"
```
Comment (KhaledSherifSayed) [inline app/src/main/res/layout/activity_main.xml]: why android:visibility="visible"? it's already visible ?!

File: app/src/main/res/navigation/navigation.xml [hunk 2]
```diff
@@ -0,0 +1,6 @@
+<?xml version="1.0" encoding="utf-8"?>
+<navigation xmlns:android="http://schemas.android.com/apk/res/android"
+    xmlns:app="http://schemas.android.com/apk/res-auto"
+    android:id="@+id/navigation">
```
Comment (KhaledSherifSayed) [inline app/src/main/res/navigation/navigation.xml]: add meaningful id

File: app/src/main/res/layout/activity_main.xml (hunk 1 above)
Comment (OmarMarei95) [reply app/src/main/res/layout/activity_main.xml]: yes, it's unnecessary attribute. already removed

File: app/src/main/res/navigation/navigation.xml (hunk 2 above)
Comment (OmarMarei95) [reply app/src/main/res/navigation/navigation.xml]: modified

**Review by KhaledSherifSayed — APPROVED** [review]

**Review by EslamHamouda — APPROVED** [review]

**Pull request merged**